tasks/main.yml

```diff
@@ -52,121 +49,55 @@
   copy:
     src: "{{ item.src }}"
     dest: "{{ item.dest }}"
-  with_items: "{{ automation_extra_files }}"
-  when: automation_extra_files is defined
-
-#
-# Install Replicated
-#
-# Parameters are:
-#   - private_address
-#   - public_address
-#   - proxy
-#   - docker_version
-#   - install replicated|operator; replicated is the default
-#   - install_url overwrite the default install url
-#
-
-- name: Set install parameters
+  with_items: "{{ replicated_automation_extra_files }}"
+  when: replicated_automation_extra_files is defined
+
+###
+### Install Replicated
+###
+### Parameters are:
+###   - replicated_private_address
+###   - replicated_public_address
+###   - replicated_proxy
+###   - replicated_docker_version
+###   - replicated_install replicated|operator; replicated is the default
+###
+
+###
+### Run the docker install for the operating system
+###
+
+- name: Check if Docker is already installed
+  stat:
+    path: /usr/bin/dockerd
+  register: docker_installed
+
+- name: Install Docker
+  include: docker/pre/{{ ansible_distribution }}.yml
+  when: docker_installed.stat.exists == False
+
+###
+### Install Replicated
+###
+
+- name: Installing Replicated Containers
+  include: scheduler/replicated/replicated.yml
+  when: replicated_install is not defined or replicated_install == "replicated"
+
+###
+### Install Replicated Operator
+###
+
+- fail:
+    msg: "Operator only install missing required replicated_daemon_address parameter"
+  when: replicated_daemon_address is not defined and replicated_install is defined and replicated_install == "operator"
+
+- name: Prep operator install env vars
   set_fact:
-    opt_private_address: '{{ private_address | default(ansible_default_ipv4.address) }}'
-    opt_public_address: '{{ public_address | default(ansible_default_ipv4.address) }}'
-    opt_proxy: '{{ proxy | default("no-proxy") }}'
-    opt_install: '{{ install | default("replicated") }}'
+    replicated_daemon_endpoint: '{{ replicated_daemon_address | default(ansible_default_ipv4.address) }}:9879'
```

Comment:
With IPv6, `replicated_daemon_address` will need to be in square brackets.   This might a TODO for a future revision.